lib/Parse/ParseDecl.cpp

```diff
@@ -8980,6 +8980,18 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
       if (ArgParams.isNull() || ArgParams.hasCodeCompletion())
         return ParserStatus(ArgParams);
     }
+    
+    // See if there are generic params.
+    auto genericParams = maybeParseGenericParams()
```

Comment:
Can you please add tests for invalid generic clause like 
```swift
case someCase<Partial
case someOtherCase<subscript>(param: subscript)
```

It seems like `maybeParseGenericParams` may emit other diagnostics when the generic parmeters is invalid, which will be removed entirely when the generic clause is gone anyways.